library/management/chat/src/main/java/org/quiltmc/qsl/chat/api/MessageChainLookup.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 The Quilt Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.chat.api;
+
+import java.util.WeakHashMap;
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.network.message.MessageChain;
+
+/**
+ * Provides a method to get a {@link ChatSecurityRollbackSupport} from a {@link MessageChain.Packer} or {@link MessageChain.Unpacker}.
+ * This class has been provided as API in the case that a user wishes to replace or extend the builtin chat signing system.
```

Comment:
Dumb question: Why?
When might a mod want to replace/extend the chat signing system?

Reply:
Hm i think that might be some off docs on my part

essentially there isnt any way to automatically track custom (un)packers since theyre a `FunctionalInterface`, so the idea was that if you create a custom (un)packer for *any* reason, i.e a custom verification system (e2e) or just adding logging, other small tweaks, ect, you could easily register it here

the map has to exist anyways, so i didnt see much reason to not just, allow people to register custom ones easily